lib/output.js

```diff
@@ -784,7 +784,9 @@ function OutputStream(options) {
         output.with_parens(function(){
             self.condition.print(output);
         });
-        output.semicolon();
+        if (output.option("beautify") && output.option("screw_ie8")) {
+            output.semicolon();
```

Comment:
I'm okay with that behavior but 2.7.5 and 2.8.7 is more forgiving with respect to `do-while` output.

With this PR users will have to use `--support-ie8` if they want to the code to run on IE8, which is reasonable.